Build error: [commonjs--resolver] Missing "#" specifier when using href="#" in SVG tags

### Environment

│ ╭── Nuxt project info (copied to clipboard) ───────────╮
│ │                                                      │
│ │  Operating system      macOS 25.3.0                  │
│ │  CPU                   Apple M1 Pro (10 cores)       │
│ │  Node.js version       v24.3.0                       │
│ │  nuxt/cli version      3.34.0                        │
│ │  Package manager       npm 11.4.2                    │
│ │  Nuxt version          4.4.2                         │
│ │  Nitro version         2.13.3                        │
│ │  Builder               vite 7.3.2                    │
│ │  Config                compatibilityDate, devtools   │
│ │  Modules               -                             │
│ │                                                      │
│ ╰──────────────────────────────────────────────────────╯

### Reproduction
Reproduction repository: https://github.com/miguel-siqueira-reis/nuxt-svg-hashtag-bug

### Steps to reproduce
1. Clone the repo.
2. Run `npm install`.
3. Run `npx nuxi build`.
4. See the error: `[commonjs--resolver] Missing "#" specifier in "nuxt" package`.

### Describe the bug
When using a literal `#` in an SVG attribute (like `href="#"` or `xlink:href="#"`) inside a Vue component, the Nuxt build fails. 

The Vite/Rollup bundler incorrectly attempts to resolve the `#` character as a Node.js Subpath Import (specifier) instead of treating it as a static SVG fragment identifier. Since Nuxt uses the `#` prefix for internal aliases (e.g., `#app`), the resolver throws a `Missing "#" specifier` error.

### Additional context
This only happens during the production build (`nuxi build`), not in development mode. It seems the static analysis of the SVG template is too aggressive, confusing standard SVG syntax with Nuxt's internal module mapping.

**Temporary Workaround:**
Using Vue's dynamic binding `:href="'#'"` bypasses the static analysis and solves the issue, but it shouldn't be necessary for valid SVG code.

### Logs

```shell-script
> build
> nuxt build

┌  Building Nuxt for production...
│
●  Nuxt 4.4.2 (with Nitro 2.13.3, Vite 7.3.2 and Vue 3.5.33)
│
●  Nitro preset: node-server
ℹ Building client...                                                 11:11:37
ℹ vite v7.3.2 building client environment for production...          11:11:37

[11:11:37]  WARN  [plugin nuxt:module-preload-polyfill] Sourcemap is likely to be incorrect: a plugin (nuxt:module-preload-polyfill) was used to transform files, but didn't generate a sourcemap for the transformation. Consult the plugin documentation for help

ℹ ✓ 44 modules transformed.                                          11:11:37

 ERROR  ✗ Build failed in 210ms                                       11:11:37

│
■  Nuxt build error: Error: [commonjs--resolver] Missing "#" specifier in "nuxt" package
```

--- a/packages/compiler-sfc/__tests__/templateTransformAssetUrl.spec.ts
+++ b/packages/compiler-sfc/__tests__/templateTransformAssetUrl.spec.ts
@@ -150,7 +150,6 @@ describe('compiler sfc: transform asset url', () => {
     expect(code).toContain(`import _imports_0 from './foo%.png'`)
   })
 
-  // #14753
   test('should not transform hash fragments on <image>', () => {
     // `<image href="#...">` is an in-document fragment reference to another
     // SVG element (like `<use>`), not a Node.js subpath import specifier.

--- a/packages/compiler-sfc/__tests__/templateTransformAssetUrl.spec.ts
+++ b/packages/compiler-sfc/__tests__/templateTransformAssetUrl.spec.ts
@@ -150,6 +150,30 @@ describe('compiler sfc: transform asset url', () => {
     expect(code).toContain(`import _imports_0 from './foo%.png'`)
   })
 
+  // #14753
+  test('should not transform hash fragments on <image>', () => {
+    // `<image href="#...">` is an in-document fragment reference to another
+    // SVG element (like `<use>`), not a Node.js subpath import specifier.
+    const { code } = compileWithAssetUrls(
+      `<svg><image href="#" /><image href="#myClip" /></svg>`,
+      { includeAbsolute: true },
+    )
+    expect(code).toContain(`href: "#"`)
+    expect(code).toContain(`href: "#myClip"`)
+    expect(code).not.toContain(`from '#'`)
+    expect(code).not.toContain(`from '#myClip'`)
+  })
+
+  test('should not transform bare `#` value into an import', () => {
+    // A bare `#` is never a valid module specifier and should always be left
+    // untouched, even on tags that otherwise support subpath imports.
+    const { code } = compileWithAssetUrls(`<img src="#" />`, {
+      includeAbsolute: true,
+    })
+    expect(code).toContain(`src: "#"`)
+    expect(code).not.toContain(`from '#'`)
+  })
+
   test('should allow for full base URLs, with paths', () => {
     const { code } = compileWithAssetUrls(`<img src="./logo.png" />`, {
       base: 'http://localhost:3000/src/',

--- a/packages/compiler-sfc/src/template/transformAssetUrl.ts
+++ b/packages/compiler-sfc/src/template/transformAssetUrl.ts
@@ -35,13 +35,14 @@ export interface AssetURLOptions {
   tags?: AssetURLTagConfig
 }
 
-// Built-in attrs that always represent resource URLs. `use` is intentionally
-// omitted because its hash-only values may still be fragment references.
+// Built-in attrs that always represent resource URLs. `use` and `image` are
+// intentionally omitted because their hash-only values are fragment references
+// to in-document elements (e.g. `<image href="#myClip" />`), not module
+// specifiers.
 const resourceUrlTagConfig: AssetURLTagConfig = {
   video: ['src', 'poster'],
   source: ['src'],
   img: ['src'],
-  image: ['xlink:href', 'href'],
 }
 
 export const defaultAssetUrlOptions: Required<AssetURLOptions> = {
@@ -49,6 +50,7 @@ export const defaultAssetUrlOptions: Required<AssetURLOptions> = {
   includeAbsolute: false,
   tags: {
     ...resourceUrlTagConfig,
+    image: ['xlink:href', 'href'],
     use: ['xlink:href', 'href'],
   },
 }
@@ -125,6 +127,8 @@ export const transformAssetUrl: NodeTransform = (
       if (
         isExternalUrl(urlValue) ||
         isDataUrl(urlValue) ||
+        // a bare `#` is never a valid import specifier
+        urlValue === '#' ||
         (isHashOnlyValue && !canTransformHashImport(node.tag, attr.name)) ||
         (!options.includeAbsolute && !isRelativeUrl(urlValue))
       ) {

